api/v1alpha1/serviceclaim_types.go

```diff
@@ -20,22 +20,41 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+type ServiceClaimApplication struct {
+	// API version of the referent.
+	APIVersion string `json:"apiVersion"`
+	// Kind of the referent.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
+	Kind string `json:"kind"`
+	// Name of the referent.
+	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
+	Name string `json:"name,omitempty"`
+	// Selector is a query that selects the workload or workloads to bind the service to
+	Selector *metav1.LabelSelector `json:"selector,omitempty"`
+}
+
+type ServiceClaimApplicationContext struct {
+	Cluster   string `json:"cluster"`
+	Namespace string `json:"namespace"`
+}
 
 // ServiceClaimSpec defines the desired state of ServiceClaim
 type ServiceClaimSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
-
-	// Foo is an example field of ServiceClaim. Edit serviceclaim_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	Name                          string                         `json:"name"`
+	ServiceClassIdentity          string                         `json:"serviceClassIdentity"`
+	ServiceEndpointDefinitionKeys string                         `json:"serviceEndpointDefinitionKeys,omitempty"`
```

Comment:
These two fields should not be strings. See registered service struct. Lets re-use the structs define for SCI and SED

Reply:
I am working on this change.

Reply:
I have added changes from PR #15. Once that PR is merged, I will rebase this PR branch.

Reply:
rebased.